go/libkb/merkle_client.go

```diff
@@ -1852,19 +1930,38 @@ func (mc *MerkleClient) lookupLeafHistorical(m MetaContext, leafID keybase1.User
 	}
 	paramer(&q)
 
-	if path, ss, apiRes, err = mc.lookupPathAndSkipSequenceTeam(m, q, currentRoot, opts); err != nil {
+	if apiRes, currentRoot, err = mc.lookupLeafAndPathHelper(m, q, nil, currentRoot, opts); err != nil {
 		return nil, nil, nil, err
 	}
 
-	if err = checker(path); err != nil {
+	path, err = mc.readPathFromAPIRes(m, apiRes, opts)
+	if err != nil {
 		return nil, nil, nil, err
 	}
-
-	opts.historical = true
-	err = mc.verifySkipSequenceAndRoot(m, ss, path.root, currentRoot, apiRes, opts)
+	resSeqno, err := apiRes.Body.AtKey("root").AtKey("seqno").GetInt64()
 	if err != nil {
 		return nil, nil, nil, err
 	}
+	if keybase1.Seqno(resSeqno) == *currentRoot.Seqno() {
+		path.root = currentRoot
+	} else {
```

Comment:
Style: too much nesting

Reply:
(See https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88 OR https://dave.cheney.net/2020/02/23/the-zen-of-go ("Return early rather than nesting deeply"))